autoinstall/FreeBSD/installerconfig

```diff
@@ -0,0 +1,129 @@
+
+debug=yes
+nonInteractive=yes
+
+{% if boot_disk_controller == 'nvme' %}
+PARTITIONS="nvd0 gpt"
+{% elif boot_disk_controller == 'sata' or boot_disk_controller == 'ide' %}
+PARTITIONS="ada0 gpt"
+{% else %}
+PARTITIONS="da0 gpt"
+{% endif %}
+DISTRIBUTIONS="kernel.txz base.txz kernel-dbg.txz lib32.txz src.txz ports.txz"
+
+#!/bin/sh
+# Set hostname
+sysrc hostname="FreeBSD-{{ hostname_timestamp }}"
+
+{% if boot_disk_controller == 'nvme' %}
+gpart bootcode -b /boot/pmbr -p /boot/gptboot -i 1 nvd0
+{% elif boot_disk_controller == 'sata' or boot_disk_controller == 'ide' %}
+gpart bootcode -b /boot/pmbr -p /boot/gptboot -i 1 ada0
+{% else %}
+gpart bootcode -b /boot/pmbr -p /boot/gptboot -i 1 da0
+{% endif %}
+
+# Set Time Zone to UTC
+echo "Setting Time Zone to UTC ..."
+/bin/cp /usr/share/zoneinfo/UTC /etc/localtime
+/usr/bin/touch /etc/wall_cmos_clock
+/sbin/adjkerntz -a
+
+echo "Set network interface with DHCP IP assignment ..." > /dev/ttyu0
+ifdev=$(ifconfig | grep '^[a-z]' | cut -d: -f1 | head -n 1)
+echo "Get ifname ${ifdev}" > /dev/ttyu0
+sysrc ifconfig_${ifdev}=DHCP
+
+# Get DHCP for nic0
+echo "Get IP with dhclient ..." > /dev/ttyu0
+dhclient ${ifdev}
+sleep 15
+echo "Check network ..." > /dev/ttyu0
+ifconfig > /dev/ttyu0
+
+# Set Proxy. without the proxy, maybe it will take more time to download pakages xorg & gnome
+{% if http_proxy_vm is defined and http_proxy_vm %}
+setenv HTTP_PROXY {{ http_proxy_vm }}
+{% endif %}
+
+# Installing packages
+echo "Installing packages ..." > /dev/ttyu0
+env ASSUME_ALWAYS_YES=YES pkg bootstrap -y
+env ASSUME_ALWAYS_YES=YES pkg update -f
+
+# Hit issue: reset by peer during install packages
+# The open-vm-tools is not installed by default
+mkdir -p /usr/local/etc/pkg/repos
+mount > /dev/ttyu0
+cp -rf /dist/packages/repos/FreeBSD_install_cdrom.conf /usr/local/etc/pkg/repos/FreeBSD_install_cdrom.conf
+env ASSUME_ALWAYS_YES=YES pkg update -f > /dev/ttyu0
+packages_to_install='bash sudo xorg kde5'
+for package_to_install in $packages_to_install
+do
+    ret=1
+    try_count=1
+    until [ $ret -eq 0 ]
+    do
+        echo "Install package $package_to_install (try $try_count time) ..." > /dev/ttyu0
+        env ASSUME_ALWAYS_YES=YES pkg install -y $package_to_install
+        ret=$?
+        try_count=$((try_count+1))
+    done
+    echo "The package($package_to_install) is already installed" > /dev/ttyu0
+done
+
+# Disable ISO repo and enable default repo
+rm -rf /usr/local/etc/pkg/repos/FreeBSD_install_cdrom.conf
+env ASSUME_ALWAYS_YES=YES pkg update -f > /dev/ttyu0
+packages_to_install='sddm open-vm-tools xf86-video-vmware xf86-input-vmmouse wget curl'
```

Comment:
1) 32bit and 64bit image have different packages.
2) For some packages with small size, it would be better to download from offical repo in case that the future released ISO doesn't include the package.  It's also fast to download packages with small size from offcial repo .  From the test result, only xorg / kde / gnome packages take much time to download from internet.

Thanks
Yuhua Zou

Reply:
Which 32bit image did you check? I checked FreeBSD 12.4 and 13.2 32bit image, I can see kde5 package in their ISO image.

Reply:
The newest 13.2 RC3 have required KDE5 in ISO.